modules/@apostrophecms/schema/ui/apos/components/AposInputArray.vue

```diff
@@ -244,6 +246,34 @@ export default {
       if (this.field.max && value.length > this.field.max) {
         return 'max';
       }
+      if (value.length && this.field.fields && this.field.fields.add) {
+        const [ uniqueFieldName, uniqueFieldSchema ] = Object.entries(this.field.fields.add).find(([ , subfield ]) => subfield.unique);
+        if (uniqueFieldName) {
+          const duplicates = this.next
+            .map(item =>
+              Array.isArray(item[uniqueFieldName])
+                ? item[uniqueFieldName].map(i => i._id).sort().join('|')
+                : item[uniqueFieldName])
+            .filter((item, index, array) => array.indexOf(item) !== index);
+
+          if (duplicates.length) {
+            duplicates.forEach(duplicate => {
+              this.items.forEach(item => {
+                uniqueFieldSchema.type === 'relationship'
+                  ? item.schemaInput.data[uniqueFieldName] && item.schemaInput.data[uniqueFieldName].forEach(datum => {
+                    item.schemaInput.fieldState[uniqueFieldName].duplicate = duplicate.split('|').find(i => i === datum._id);
+                  })
+                  : item.schemaInput.fieldState[uniqueFieldName].duplicate = item.schemaInput.data[uniqueFieldName] === duplicate;
+              });
+            });
+
+            return {
+              name: 'duplicate',
+              message: `duplicate ${this.$t(uniqueFieldSchema.label) || uniqueFieldName}`
```

Comment:
If you are going to go the road of localizing here, then you need to localize "duplicate" too

Reply:
We need to draw the user's attention to the specific array item that contains the duplicate. A new prop on the draggable would work for that in the inline case.

Reply:
Ok, done localizing the error message. But the error name has to be "duplicate" in lowercase because A3 automatically creates a css class based on it and I use it to prevent every subfield to be marked as in error.

```css

::v-deep .apos-field--array.apos-field--error-duplicate {
    .apos-input {
      border-color: var(--a-base-8);
    }
}
```